protos/flyteidl/plugins/sagemaker/parameter_ranges.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package flyteidl.plugins.sagemaker;
+
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins";
+
+// HyperparameterScalingType defines the way to increase or decrease the value of the hyperparameter
+// https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning-define-ranges.html
+enum HyperparameterScalingType {
+    AUTO = 0;
+    LINEAR = 1;
+    LOGARITHMIC = 2;
+    REVERSELOGARITHMIC = 3;
+}
+
+// ContinuousParameterRange refers to a continuous range of hyperparameter values, allowing
+// users to specify the search space of a floating-point hyperparameter
+message ContinuousParameterRange {
+    double max_value = 1;
+    double min_value = 2;
+    HyperparameterScalingType scaling_type = 3;
+}
+
+// IntegerParameterRange refers to a discrete range of hyperparameter values, allowing
+// users to specify the search space of an integer hyperparameter
+message IntegerParameterRange {
+    int64 max_value = 1;
+    int64 min_value = 2;
+    HyperparameterScalingType scaling_type = 3;
+}
+
+// ContinuousParameterRange refers to a continuous range of hyperparameter values, allowing
+// users to specify the search space of a floating-point hyperparameter
+message CategoricalParameterRange {
+    repeated string values = 1;
+}
+
+message ParameterRangeOneOf {
+    oneof parameter_range_type {
+        ContinuousParameterRange continuous_parameter_range = 1;
+        IntegerParameterRange integer_parameter_range = 2;
+        CategoricalParameterRange categorical_parameter_range = 3;
+    }
+}
+
+// ParameterRanges is a map that maps hyperparameter name to the corresponding hyperparameter range
+message ParameterRanges {
+    map<string, ParameterRangeOneOf> parameter_range_map = 1;
+}
```

Comment:
nit:
```suggestion
}

```

Reply:
To summarize the issue: if the enum is not wrapped inside a message, the generated python code for an enum, say,
```proto
enum FixedRateUnit {
    MINUTE = 0;
    HOUR = 1;
    DAY = 2;
}
```
will be orphans exposed at the module level
```python
_schedule_pb2.MINUTE
```
https://github.com/lyft/flytekit/blob/master/flytekit/models/schedule.py#L10

When we have multiple of them in a proto file, you can see that the generated python file can be confusing, especially when there are name collisions such as 
```
enum A {
   AUTO = 0;
   OFF = 1;
}

enum B {
  LINEAR = 0;
  AUTO = 1;
}
```
So the way to resolve this issue is to wrap every enum inside a message